src/main/java/core/basesyntax/figures/Figure.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.figures;
+
+public abstract class Figure implements Drawable {
+    protected String color;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    protected abstract double calculateArea();
```

Comment:
Don't use abstract classes to set behavior for classes. In this case, the `calculateArea` method could potentially be part of an interface, as it seems to define a contract that must be fulfilled by all concrete `Figure` subclasses.

Reply:
As far as method is only used inside each figure and is not part of public API, it is hidden as protected inside of the abstract class.